lib/p2p/Peer.ts

```diff
@@ -255,6 +422,102 @@ class Peer extends EventEmitter {
 
     this.emit('error', { msg, err });
   }
+
+  private sendHello = (): HelloPacket => {
+    // TODO: use real values
+    const packet = new HelloPacket({
+      version: '123',
+      nodeKey: '123',
+      listenPort: 20000,
+      pairs: ['BTC/LTC'],
+    });
+
+    this.sendPacket(packet);
+
+    return packet;
+  }
+
+  private handleHello = (packet: HelloPacket): void => {
+    this.handshakeState = { ...this.handshakeState, ...packet.body };
+
+    // TODO: define XUD_MIN_VERSION, throw exception if version below
+  }
+
+  private sendPing = (): PingPacket => {
+    this.lastPingId = uuidv1();
+    const packet = new PingPacket({ id: this.lastPingId! });
+
+    this.sendPacket(packet);
+
+    return packet;
+  }
+
+  private handlePing = (packet: PingPacket): void  => {
+    const { id } = packet.body;
+
+    if (!id) {
+      return;
+    }
+
+    this.sendPong(id + 1);
+  }
+
+  private sendPong = (id: string): PongPacket => {
+    const packet = new PongPacket({ id });
+
+    this.sendPacket(packet);
+
+    return packet;
+  }
+
+  private handlePong = (packet: PongPacket): boolean => {
+    const { id } = packet.body;
+    if (!this.lastPingId) {
+      this.logger.debug(`Peer sent an unsolicited pong (${this.id})`);
+      return false;
+    }
+
+    if (id !== this.lastPingId) {
+      this.logger.debug(`Peer sent an invalid pong id (${this.id})`);
+      return false;
+    }
+
+    this.lastPingId = null;
+    return true;
+  }
+}
+
+class ResponseEntry {
```

Comment:
The name for this class threw me off a bit. As I understand it this class represents a wait for a response from a peer. In that case I think a name like `PendingResponse` might be better.

Reply:
Yes, it wasn't a brilliant name.
But i'm not sure about `PendingReponse`, because it doesn't represent a response. 

I changed it to `PendingResponseEntry` for now. Let me know if you have a better name.

Reply:
Is my understanding above correct tho? I am also struggling to come up with a succint name, feels like English is missing a word. Some other ideas, none of which are perfect, `ResponseTimer`, `WaitForResponse` (wait in the noun sense, `AwaitingResponse`.